src/arithmetic.jl

```diff
@@ -73,88 +70,119 @@ function ones{T<:Number}(a::HomogeneousPolynomial{T}, order::Int)
     v = Array{HomogeneousPolynomial{T}}(order+1)
     @simd for ord in eachindex(v)
         @inbounds num_coeffs = size_table[ord]
-        vT = Array{T}(num_coeffs)
-        @simd for ind in 1:num_coeffs
-            @inbounds vT[ind] = one(a.coeffs[1])
-        end
-        @inbounds v[ord] = HomogeneousPolynomial(vT, ord-1)
+        @inbounds v[ord] = HomogeneousPolynomial(ones(T, num_coeffs), ord-1)
     end
     return v
 end
 
 ones{T<:Number}(::Type{HomogeneousPolynomial{T}}, order::Int) =
     ones( HomogeneousPolynomial([one(T)], 0), order)
 
-zero{T<:Number}(a::TaylorN{T}) = TaylorN(zero(T), a.order)
-one{T<:Number}(a::TaylorN{T}) = TaylorN(one(T), a.order)
+zero(a::TaylorN) = TaylorN(zero(a.coeffs[1]), a.order)
+one(a::TaylorN) = TaylorN(one(a.coeffs[1]), a.order)
 
 
 
 ## Addition and substraction ##
 for f in (:+, :-)
-    for T in (:Taylor1, :HomogeneousPolynomial, :TaylorN)
+    dotf = ifelse( f == :+, :(.+), :(.-))
+    for T in (:Taylor1, :TaylorN)
         @eval begin
             ($f){T<:Number,S<:Number}(a::$T{T}, b::$T{S}) = $f(promote(a,b)...)
             function $f{T<:Number}(a::$T{T}, b::$T{T})
-                a, b = fixorder(a, b)
-                v = similar(a.coeffs)
-                @simd for i in eachindex(v)
-                    @inbounds v[i] = $f(a.coeffs[i], b.coeffs[i])
+                aord = a.order
+                bord = b.order
+                ac = view(a.coeffs, :)
```

Comment:
Not really needed, but the expressions were getting too long. Maybe with `la` things are easier to read.